src/emmc.c

```diff
@@ -99,11 +100,26 @@ gboolean r_emmc_write_bootpart(const gchar *device, gint bootpart_active, GError
 		goto out;
 	}
 
+	ret = r_emmc_read_extcsd(fd, extcsd);
+	if (ret) {
+		g_set_error(error, R_UPDATE_ERROR, R_UPDATE_ERROR_FAILED,
+				"Could not read from extcsd register %d in %s\n",
+				EXT_CSD_PART_CONFIG, device);
+		goto out;
+	}
+
+	/* Keep BOOT_ACK value as it is. Resetting this bit might prevent
+	 * proper boot process on some platforms since in some SoCs, ROM boot
+	 * loaders need this flag to be set.
+	 * The PARTITION_ACCESS value is handled inside the kernel so we do not
+	 * need to take care of it here. */
+	value = extcsd[EXT_CSD_PART_CONFIG] & ~(1 << 6);
```

Comment:
This clears `BOOT_ACK` which is the opposite of what we intended to achieve.

Shouldn't that be:
``` suggestion
	value = extcsd[EXT_CSD_PART_CONFIG] & (1 << 6);
```